core/base/document.php

```diff
@@ -1590,10 +1590,13 @@ public function process_element_import_export( Controls_Stack $element, $method,
 
 			// Do not add default value to the final settings, if there is no value at the
 			// data before the methods `on_import` or `on_export` called.
-			$has_value = isset( $element_data[ $control['name'] ] );
+			$has_value = isset( $element_data['settings'][ $control['name'] ] );
 
 			if ( $has_value && method_exists( $control_class, $method ) ) {
-				$element_data['settings'][ $control['name'] ] = $control_class->{$method}( $element->get_settings( $control['name'] ), $control );
+				$element_data['settings'][ $control['name'] ] = $control_class->{$method}(
+					$element_data['settings'][ $control['name'] ],
+					$control
+				);
```

Comment:
Maybe add some tests on the 'process_element_import_export' method?

Reply:
The test I wrote is integration test that walk through this method